File: app/src/main/java/com/ne1c/gitteroid/models/data/UserModel.java
package com.ne1c.gitteroid.models.data;

import android.os.Parcel;
import android.os.Parcelable;

public class UserModel implements Parcelable {
    public String id = "";
    public String username = "";
    public String displayName = "";
    public String url = "";
    public String avatarUrlSmall = "";
    public String avatarUrlMedium = "";

    public UserModel() {
    }

    protected UserModel(Parcel in) {
        this.id = in.readString();
        this.username = in.readString();
        this.displayName = in.readString();
        this.url = in.readString();
        this.avatarUrlSmall = in.readString();
        this.avatarUrlMedium = in.readString();
    }

    public static final Parcelable.Creator<UserModel> CREATOR = new Parcelable.Creator<UserModel>() {
        public UserModel createFromParcel(Parcel source) {
            return new UserModel(source);
        }

        public UserModel[] newArray(int size) {
            return new UserModel[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.id);
        dest.writeString(this.username);
        dest.writeString(this.displayName);
        dest.writeString(this.url);
        dest.writeString(this.avatarUrlSmall);
        dest.writeString(this.avatarUrlMedium);
    }
}

File: app/src/main/kotlin/com/ne1c/gitteroid/models/data/UserModel.kt
package com.ne1c.gitteroid.models.data

import android.os.Parcel
import android.os.Parcelable

data class UserModel(var id: String = "",
                     var username: String = "",
                     var displayName: String = "",
                     var url: String = "",
                     var avatarUrlSmall: String = "",
                     var avatarUrlMedium: String = "") : Parcelable {
    constructor(source: Parcel) : this(source.readString(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readString(),
            source.readString())

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel?, flags: Int) {
        dest?.writeString(id)
        dest?.writeString(username)
        dest?.writeString(displayName)
        dest?.writeString(url)
        dest?.writeString(avatarUrlSmall)
        dest?.writeString(avatarUrlMedium)
    }

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<UserModel> = object : Parcelable.Creator<UserModel> {
            override fun createFromParcel(source: Parcel): UserModel = UserModel(source)
            override fun newArray(size: Int): Array<UserModel?> = arrayOfNulls(size)
        }
    }
}
